src/wp-includes/functions.php

```diff
@@ -3886,6 +3886,20 @@ function _default_wp_die_handler( $message, $title = '', $args = array() ) {
 		$message  .= "\n<p><a href='javascript:history.back()'>$back_text</a></p>";
 	}
 
+	// Ensure HTTP 500 status code for critical errors, even if headers were already sent.
+	$is_critical_error = false;
+	if ( is_wp_error( $message ) ) {
```

Comment:
All of the `function_exists()` checks above for `__()` and `esc_url()` make me think that actually we should maybe not assume that `is_wp_error()` exists. In any case, it isn't necessary:


```suggestion
	if ( $message instanceof WP_Error ) {
```

Reply:
fixed